app/models/gpo_confirmation.rb

```diff
@@ -1,19 +1,65 @@
 class GpoConfirmation < ApplicationRecord
   self.table_name = 'usps_confirmations'
 
+  validates :entry, presence: true
+  validates :otp, :address1, :city, :state, :zipcode, presence: true
+  validate :zipcode_is_valid
+
+  ZIP_REGEX = /^(\d{5})[-+]?(\d+)?$/
+
+  def method_missing(method, *args)
```

Comment:
Could we add a list of supported methods in here somewhere so we don't just allow list everything?

Reply:
+1 I think this is a little sneaky... especially since this is already an ActiveRecord model with a ton of dynamic methods.

I would consider something like this:
```ruby
ENTRY_ATTRIBUTES = %i[otp address1 city state zipcode]
ENTRY_ATTRIBUTES.each do |attrib|
  define_method(attrib) do
    entry[attrib]
  end
end
```
and even then I'd consider something like prefixin them with `entry_zipcode` etc to make it clearer to people searching the codebase where they come from
```ruby
ENTY_ATTRIBUTES = {
  entry_otp: :otp,
  entry_address1: :address1,
  # etc
}
ENTRY_ATTRIBUTES.each do |method_name, attribute_name|
  define_method(method_name) do
    entry[attribute_name]
  end
end
validates :entry_otp, :entry_addresss, presence: :true
```

Reply:
Thanks, I like this and felt a little dirty doing method_missing

Reply:
Fixed in [06361c9](https://github.com/18F/identity-idp/pull/9823/commits/06361c9b1f6b5c0a31a8a4e421d61d77703567e8)